README.md

```diff
@@ -381,464 +607,224 @@ class MyBaseClass extends WidgetBase<WidgetProperties> {
-
-##### Authoring a Base Theme
-
-A base theme is authored using `css-modules` and `cssnext`. The base theme `css` file should be located in a `styles` folder within the Widget's package directory.
-The `typed-css-modules` [cli](https://github.com/Quramy/typed-css-modules#cli) should be used in `watch` mode in order to generate typings for TypeScript usage. This is automatically included within the `dojo build -w` command from `dojo-cli`.
-
-```
-tabPanel
-├── createTabPanel.ts
-└── styles
-    └── tabPanel.css
-```
+##### Implementing Custom Meta
 
-The `baseClasses` css must contain a complete set of all of the classes you wish to apply to a widget as all theme and extra classes are limited by the classnames made available here.
-Classnames are locally scoped as part of building a Dojo application. A theme `key` is generated at build time to locate the themes for each class where a theme is set.
+You can create your own meta if you need access to DOM nodes.
 
-```css
-/* tabpanel.css */
-.root {
-	background: red;
-}
+```typescript
+import MetaBase from "@dojo/widget-core/meta/Base";
 
-.tab {
-	background: blue;
+class HtmlMeta extends MetaBase {
+    get(key: string): string {
+        this.requireNode(key);
+        const node = this.nodes.get(key);
+        return node ? node.innerHTML : '';
+    }
 }
 ```
 
-##### Registering `baseClasses`
-
-To apply `baseClasses` a widget must use `ThemeableMixin` and the `theme` decorator from `mixins/Themeable` to register the `baseClasses`.
+And you can use it like:
 
-```ts
-import { WidgetProperties } from '@dojo/widget-core/interfaces';
-import { WidgetBase } from '@dojo/widget-core/WidgetBase';
-import { ThemeableMixin, theme } from '@dojo/widget-core/mixins/Themeable';
-import * as baseClasses from './styles/tabpanel.css';
+```typescript
+class MyWidget extends WidgetBase<WidgetProperties> {
+    // ...
+    render() {
+        // run your meta
+        const html = this.meta(HtmlMeta).get('comment');
 
-@theme(baseClasses)
-class MyThemeableWidget extends ThemeableMixin(WidgetBase)<WidgetProperties> {
-	// ...
+        return v('div', { key: 'root', innerHTML: html });
+    }
+    // ...
 }
 ```
 
-Basic usage:
-
-```ts
-import { WidgetProperties } from '@dojo/widget-core/interfaces';
-import { WidgetBase } from '@dojo/widget-core/WidgetBase';
-import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
-import * as baseClasses from './styles/tabpanel.css';
+Meta classes are provided with a few hooks into the widget, passed to the constructor:
 
-interface MyThemeableWidgetProperties extends WidgetProperties, ThemeableProperties;
+* `nodes` - A map of `key` strings to DOM elements. Only `v` nodes rendered with `key` properties are stored.
```

Comment:
Only `v` nodes rendered with `key` properties are stored. -> reword to make this clearer